### PR TITLE
clang-tidy: mostly readability stuff

### PR DESCRIPTION
**Pull request by neheb**

### CODE REVIEW

File: hardware/CurrentCostMeterTCP.h [hunk 1]
```diff
@@ -2,13 +2,13 @@
 
 #include "CurrentCostMeterBase.h"
 
-class CurrentCostMeterTCP: public CurrentCostMeterBase
+class CurrentCostMeterTCP : public CurrentCostMeterBase
 {
-public:
-	CurrentCostMeterTCP(const int ID, const std::string &IPAddress, const unsigned short usIPPort);
+      public:
+	CurrentCostMeterTCP(int ID, const std::string &IPAddress, unsigned short usIPPort);
```
Comment (gizmocuz) [inline hardware/CurrentCostMeterTCP.h:8]: Maybe I am missing something, but what is wrong declaring a function parameter with const ?

Comment (neheb) [reply hardware/CurrentCostMeterTCP.h:8]: From https://clang.llvm.org/extra/clang-tidy/checks/readability-const-return-type.html

Such use of const is usually superfluous, and can prevent valuable compiler optimizations.

**Comment by gizmocuz** [comment]
@neheb , I think you can also take it a bit to far....?
Could you undo:
- simplify booleans, actually when you code/debug for a long period, you might overlook the exclamation mark
- use emplace_back
- use std::move

File: hardware/CurrentCostMeterTCP.h (hunk 1 above)
Comment (gizmocuz) [reply hardware/CurrentCostMeterTCP.h:8]: Thats for "const returns", I'm referring to the use of 'const' in the function parameters

CurrentCostMeterTCP(const int ID, const std::string &IPAddress, const unsigned short usIPPort);

why could we not declare 'int ID' as const ? or 'usIPPort' ?

Comment (neheb) [reply hardware/CurrentCostMeterTCP.h:8]: Ah. http://clang.llvm.org/extra/clang-tidy/checks/readability-avoid-const-params-in-decls.html

Looks like there's no real need for it.

**Comment by neheb** [comment]
Removed those.

**Comment by gizmocuz** [comment]
Thank you again !!

**Pull request merged**